packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -616,6 +616,39 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 		this[__private__dont__use].FS.rename(fromPath, toPath);
 	}
 
+	/** @inheritDoc */
+	@rethrowFileSystemError('Could not copy "{path}"')
+	cp(fromPath: string, toPath: string) {
+		const FS = this[__private__dont__use].FS;
+
+		const fromStat = FS.stat(fromPath);
+
+		const fromMode = parseInt(fromStat.mode.toString(8).substring(0, 1));
```

Comment:
Let's make this more human-readable – here's the thoughts I had when I was reading this:

* What's `mode`?
* What's `8`? 
* Why does it need to be `substring`-ed?

Reply:
https://github.com/WordPress/wordpress-playground/pull/846/files#diff-b33e1e6834fb94ac2cb5338e3ae17c60abf7ac88cb057189a1915531e55f7243R629

Reply:
Thank you for the update! I'm still a bit confused about what's a mode, what's that flag etc:

```ts
		// Emscripten directories will have the sticky-bit set
		// https://linux.die.net/man/1/chmod
		const fromIsDir = fromStat.mode & 0o40000;
```

Let's be kind to readers of this code and explain these concepts in place instead of just pointing to the linux manual.

Reply:
https://github.com/WordPress/wordpress-playground/blob/cc576fc972ceaabeba7660aaa39494c95aca2de8/packages/php-wasm/universal/src/lib/base-php.ts#L631-L639